### ISSUE TITLE
Unable to get value of the property 'prototype' in IE

### ISSUE THREAD

**Issue by abigwonderful** [swilliams/jq-wysihat#10]
I just started digging into jq-wysihat and was running some testing in IE. In all other browsers, the editor seems to be working as expected, but IE doesn't want to cooperate. 
currently getting the following error:

Message: Unable to get value of the property 'prototype': object is null or undefined
Line: 569
Char: 3
Code: 0
URI: *****example.com/js/wysihat/jq-wysihat.js

Any thoughts/suggestions?

**Comment by abigwonderful** [comment]
if it helps:
http://events.alltrips.com/simple.html

also a second error that actually occurs before the one posted above (forgot to add earlier):

User Agent: Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.2; Trident/4.0; SearchToolbar 1.2; .NET CLR 1.1.4322; .NET CLR 2.0.50727; .NET CLR 3.5.30729)
Timestamp: Mon, 31 Oct 2011 20:49:36 UTC

Message: Object expected
Line: 507
Char: 5
Code: 0
URI: http://events.alltrips.com/js/wysihat/jq-wysihat.js

**Comment by Senne** [comment]
+1 

IE8: Object Expected - line 507, character 5
IE9: Unable to get value of the property "prototype': object is null or undefined - line 569, character 3

**Comment by pezholio** [comment]
I get the same thing. This is a showstopper for me. Shame, as it's potentially a really useful tool

**Comment by pezholio** [comment]
I've fixed it (a bit hacky mind) by removing the IE fallbacks and including selection.js instead. See this gist for more info:

https://gist.github.com/2556956

**Comment by capocasa** [comment]
I did a clean fix.

Apparently, IE behaves like this:

(function () {
  window.foo = 1;
  alert(foo); // foo is undefined
});
alert(foo); // works

(function () {
  var foo = 1;
  window.foo = foo;
  alert(foo); // works
});
alert(foo); // works

This was already being done with _Wysihat_, I applied it to _Range_ and _Selection_ as well and it works.

The patch is in my bugfix fork: https://github.com/carlocapocasa/jq-wysihat

Pull request: https://github.com/swilliams/jq-wysihat/pull/13

**Comment by pezholio** [comment]
Great, thanks for that. However, it doesn't seem to work for me in Chrome. I get the error `Uncaught TypeError: Cannot read property 'prototype' of undefined` on line 514. Any ideas?

**Comment by capocasa** [comment]
Just commited the fix... does it work?

I just learned that JavaScript always executes var statements, even when they are in an if block. So my fixes var statements clobbered the Range and Selection objects with 'undefined'. Fixed now, just using straight global variables now.

(function () {
foo = 1;
alert(foo); // works
});
window.foo = foo;
alert(foo); // works
alert(window.foo); // works

**Comment by pezholio** [comment]
Yup, that seems to work perfectly - thanks! Only thing that doesn't work is the header selection stuff (in examples/dropdown.html), but I don't think that's your fault. Will try and fix and add a pull request.

**Comment by capocasa** [comment]
Nice! Merged it.
